Sources/CoreCommands/Options.swift

```diff
@@ -437,6 +437,9 @@ public struct BuildOptions: ParsableArguments {
     @Flag(help: "Enable or disable indexing-while-building feature")
     public var indexStoreMode: StoreMode = .autoIndexStore
 
+    @Flag(name: .customLong("experimental-prepare-for-indexing"))
+    var prepareForIndexing: Bool = false
```

Comment:
If we want this as an option could this be a broader one something `perform-experimental-minimal-build` or something like that which is not necessarily coupled with indexing so tightly but one of the use-cases is indexing?

Reply:
It's not really a build that can be used for anything other than indexing. The only output is .swiftmodules files.

Reply:
I suspect there might be some other uses in the future?

Reply:
We also allow errors + skip a lot that would be somewhat unexpected in more general cases. TBH I find it hard to imagine this would actually be useful for other purposes, though perhaps "only generating swiftmodules" (without the extra flags) may be? CC @kubamracek since IIUC this is somewhat similar to embedded needs?

In an ideal world this wouldn't be a separate arena and could be part of the same build, where any module compiled in this way would be rebuilt during the build, but *not* rebuilt by "preparation" if it is already up to date. Not sure if that helps with naming at all though 🤔.

Reply:
That's a great point @bnbarham about reusing modules between regular build and this preparation build. Certainly something we can work towards and probably needs a revisit of the architecture. I am also curious how we integrated embedded into all this (independent of preparation).

At any rate, I'm not sure there is a concept of the experimental experimental options :), but I'm inclined to leave it like this for now. As we get the sourcekit-lsp side lined up with this we can see if anything more natural jumps out.

Reply:
> I am also curious how we integrated embedded into all this (independent of preparation).

It isn't today, which is partially why I'm bringing it up - since maybe we should be splitting the "don't need objects" from "preparation". It's unlikely that would be a command line option in the embedded case, ie. we probably want some manifest changes. But maybe still useful to think about.

https://github.com/apple/swift/blob/main/docs/EmbeddedSwift/UserManual.md#libraries-and-modules-in-embedded-swift has a little more information there.

> Certainly something we can work towards and probably needs a revisit of the architecture.

Yeah, just to be clear, I was in no way suggesting this is what we should do now. One day :)

Reply:
I'm fine leaving it as is today but I think it's important to consider whether there is a more general thing hiding here that would allow us to cut down on the flags...

Reply:
@xedin Agreed. Certainly some of this may prove useful in the embedded case, e.g. skipping object files in the dependencies. But this mode feels like it's violating some of the architecture (like how it reaches deep down to remove the -c option). I think much more thought needs to be put into it to make it more generally applicable.